cpp/src/Slice/Parser.cpp

```diff
@@ -648,48 +648,44 @@ namespace
         if (stripMarkup)
         {
             // Strip HTML markup.
-            do
+            while ((pos = comment.find('<', pos)) != string::npos)
             {
-                pos = comment.find('<', pos);
-                if (pos != string::npos)
+                string::size_type endpos = comment.find('>', pos);
+                if (endpos == string::npos)
                 {
-                    string::size_type endpos = comment.find('>', pos);
-                    if (endpos == string::npos)
-                    {
-                        break;
-                    }
-                    comment.erase(pos, endpos - pos + 1);
+                    break;
                 }
-            } while (pos != string::npos);
+                comment.erase(pos, endpos - pos + 1);
+            }
+        }
 
```

Comment:
This logic was nested inside `if (stripMarkup)`. This doesn't seem right to me.
XML escaping should be unrelated to markdown striping.